File: code/src/test/scala/cern/colt/matrix/tlong/impl/DiagonalLongMatrix2DViewTest.scala
package cern.colt.matrix.tlong.impl

//remove if not needed
import scala.collection.JavaConversions._

class DiagonalLongMatrix2DViewTest(arg0: String) extends DiagonalLongMatrix2DTest(arg0) {

  protected def createMatrices() {
    DINDEX = 3
    A = new DiagonalLongMatrix2D(NCOLUMNS, NROWS, -DINDEX)
    DLENGTH = A.asInstanceOf[DiagonalLongMatrix2D].diagonalLength()
    A = A.viewDice()
    B = new DiagonalLongMatrix2D(NCOLUMNS, NROWS, -DINDEX).viewDice()
    Bt = new DiagonalLongMatrix2D(NROWS, NCOLUMNS, DINDEX).viewDice()
  }
}

File: src/test/java/cernx/colt/matrix/tlong/impl/DiagonalLongMatrix2DViewTest.java
package cernx.colt.matrix.tlong.impl;

public class DiagonalLongMatrix2DViewTest extends DiagonalLongMatrix2DTest {

    public DiagonalLongMatrix2DViewTest(String arg0) {
        super(arg0);
    }

    protected void createMatrices() throws Exception {
        DINDEX = 3;
        A = new DiagonalLongMatrix2D(NCOLUMNS, NROWS, -DINDEX);
        DLENGTH = ((DiagonalLongMatrix2D) A).diagonalLength();
        A = A.viewTranspose();
        B = new DiagonalLongMatrix2D(NCOLUMNS, NROWS, -DINDEX).viewTranspose();
        Bt = new DiagonalLongMatrix2D(NROWS, NCOLUMNS, DINDEX).viewTranspose();
    }

}
